### ISSUE TITLE
Server should remove interface on exit

### ISSUE THREAD

**Issue by Nuru** [99designs/aws-vault#360]
When the `aws-vault` proxy server starts on Linux, it [creates a network interface](https://github.com/99designs/aws-vault/blob/2bec5348b22f05b8ad9ae7ca3371791c471193e4/server/alias_linux.go#L7-L9) to capture traffic directed to 169.254.169.254. However, when the server exits, it does not remove that interface, and if you try to start the server a second time, it fails with exit status 2 because it cannot create the interface (because it already exists). 

When the server exits, it should remove the interface it added.

By the way, the workaround for this `exit status 2` on linux is:
```
sudo ip addr del 169.254.169.254 dev lo
```
On macOS this does not seem to be a problem, but if it is, you can remove the added alias with:
```
sudo ifconfig lo0 -alias 169.254.169.254
```
I don't know if this is an issue on WSL, but I would expect it to be, and I would expect it to be fixed with
```
netsh interface ipv4 delete address "Loopback Pseudo-Interface 1" address=169.254.169.254
```
but I have not tested it.

**Comment by Nuru** [comment]
This should be a pretty easy fix. Is there any reason not to just automatically delete the interface on server exit? 

Alternately, to handle `kill -9` situations, the server could try to use the interface if it exists, or print out a better error message explaining how to resolve the problem.

**Comment by Nuru** [comment]
This appears to still be an issue in version 5.3.2. How hard is this to fix?

**Comment by mtibben** [comment]
Fixed in #627
